### PR TITLE
Homepage -> Navbar -> SearchBox

### PR DESCRIPTION
**Pull request by emrerdem1**
This PR only includes `SearchBox `components and its stylings that is simplified in terms of media queries.

Kindly let me know if you see anything weird. :)

### CODE REVIEW

**Comment by emrerdem1** [comment]
PTAL.

File: src/components/config/i18next.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import i18n from "i18next";
+import { initReactI18next } from "react-i18next";
+import translationAR from "../locales/ar/translation.json";
+import translationTR from "../locales/tr/translation.json";
```
Comment (louisrli) [inline src/components/config/i18next.js]: hmm why are these files here too in the searchbox one? maybe you need to pull?

Comment (emrerdem1) [reply src/components/config/i18next.js]: I did pull before pushing this, that's why it came here too I believe. I did pull and solve conflicts so that you can directly merge to `dev` if everything is okay.

**Comment by emrerdem1** [comment]
PTAL

File: src/components/layout/nav/search/Search.jsx [hunk 2]
```diff
@@ -0,0 +1,156 @@
+import React, { useState } from "react";
+import Col from "react-bootstrap/Col";
+import Button from "react-bootstrap/Button";
+import Form from "react-bootstrap/Form";
+import "./Search.scss";
+
+//This will be serving as a temporary database until we build the real one completely.
```
Comment (louisrli) [inline src/components/layout/nav/search/Search.jsx]: as mentioned on the other change i think since omar is also using this you guys should move this out into a common file and one person PRs it

Comment (emrerdem1) [reply src/components/layout/nav/search/Search.jsx]: Yes, I'm gonna `import` it from Omar's if he merged it.

Comment (louisrli) [reply src/components/layout/nav/search/Search.jsx]: please do a separate small PR either you or omar do it, small changes are better, it should be a quick approval with the new file. in a common file.

**Review by MiraAr — APPROVED** [review]

File: src/components/layout/nav/search/Search.jsx [hunk 3]
```diff
@@ -1,7 +1,156 @@
-import React from "react";
+import React, { useState } from "react";
```
Comment (louisrli) [inline src/components/layout/nav/search/Search.jsx:1]: why are the navbar test files deleted above?

File: src/components/layout/nav/search/Search.jsx [hunk 4]
```diff
@@ -1,7 +1,156 @@
...
+  },
+];
 
 const Search = () => {
-  return <div></div>;
+  // It is left empty because we want search to start as empty.
+  const [searchQueryContainer, setSearchQuery] = useState("");
+  const [showEmptySearchWarning, setEmptySearchWarning] = useState(false);
+
+  const onlyFirstImageOfProduct = 0;
```
Comment (louisrli) [inline src/components/layout/nav/search/Search.jsx]: in the technical manual, i think i included something about magic numbers. these are usually CONSTANTS_LIKE_THIS. please check the technical manual and see if you can find the related reading and make the appropriate change

File: src/components/layout/nav/search/Search.jsx [hunk 5]
```diff
@@ -1,7 +1,156 @@
...
+      "https://fgl.scene7.com/is/image/FGLSportsLtd/FGL_332654173_01_a-Nike-Mens-Flex-Control-3-Training-Shoes-Black-Anthracite-AJ5911-002?wid=800&hei=800&bgColor=0,0,0,0&fmt=png-alpha&resMode=sharp2&op_sharpen=1",
+      "https://fgl.scene7.com/is/image/FGLSportsLtd/FGL_332762662_10_a-Nike-Womens-Air-Precision-II-Basketball-Shoes-White-Black-AA7069-100?bgColor=0,0,0,0&fmt=png-alpha&hei=528&resMode=sharp&qlt=85,1&op_sharpen=1",
+    ],
+  },
+];
 
 const Search = () => {
-  return <div></div>;
+  // It is left empty because we want search to start as empty.
+  const [searchQueryContainer, setSearchQuery] = useState("");
```
Comment (louisrli) [inline src/components/layout/nav/search/Search.jsx]: usually the convention here for `useState` is that const [x, setX] like this, therefore the two should be the same (e.g., `[searchQuery, setSearchQuery]`)

File: src/components/layout/nav/search/Search.jsx [hunk 6]
```diff
@@ -1,7 +1,156 @@
...
+  const [showEmptySearchWarning, setEmptySearchWarning] = useState(false);
+
+  const onlyFirstImageOfProduct = 0;
+  const amountOfProductToShowInSearch = 3;
+
+  // This will be sending input value to database with fetching if input is not empty when "Search" button is clicked.
```
Comment (louisrli) [inline src/components/layout/nav/search/Search.jsx:18]: generally we do this:

```
// TODO(mygithubname): Link this to the database
```

this makes it clear that this is code still left to be done. i think this code should link to the backend eventually right? you can still keep your comment above though, it's good

File: src/components/layout/nav/search/Search.jsx [hunk 7]
```diff
@@ -1,7 +1,156 @@
...
+      return;
+    } else {
+      setEmptySearchWarning(false);
+    }
+  };
+
+  // This will setting input values onChange and will show preview card only if there is something typed.
+  const handleSearchChanges = (e) => {
+    setSearchQuery(e.target.value);
+    if (!searchQueryContainer) {
```
Comment (louisrli) [inline src/components/layout/nav/search/Search.jsx]: hm. i see some copy pasted code lines 81-86 92-97

File: src/components/layout/nav/search/Search.jsx [hunk 8]
```diff
@@ -1,7 +1,156 @@
...
+      setEmptySearchWarning(true);
+      return;
+    } else {
+      setEmptySearchWarning(false);
+    }
+  };
+
+  const handleSearchPreview = () => {
+    const previewItemsContainer = mockDatabase
+      .filter((product) => product.tags.includes(searchQueryContainer))
```
Comment (louisrli) [inline src/components/layout/nav/search/Search.jsx]: is tags a string or an array?

File: src/components/layout/nav/search/Search.jsx [hunk 9]
```diff
@@ -1,7 +1,156 @@
...
+          <div className="brand">{product.brand}</div>
+          <div className="price">
+            {product.currency}
+            {product.price}
+          </div>
+        </Col>
+      ));
+
+    // This will be more precise to show if it's still loading or failed as searching when we implement Fetching.
+    if (previewItemsContainer.lenght === 0) {
```
Comment (louisrli) [inline src/components/layout/nav/search/Search.jsx]: typo here (p.s. this is why we write tests ;) if you're relying on a human to find this bug later, then that's exatctly a scenario we'd want to write tests in)

File: src/components/layout/nav/search/Search.jsx [hunk 10]
```diff
@@ -1,7 +1,156 @@
...
+      return <span>It's still loading.</span>;
```
Comment (louisrli) [inline src/components/layout/nav/search/Search.jsx]: https://getbootstrap.com/docs/4.4/components/spinners/

File: src/components/layout/nav/search/Search.jsx [hunk 11]
```diff
@@ -1,7 +1,156 @@
...
+          <div className="title">{product.title}</div>
+          <div className="brand">{product.brand}</div>
+          <div className="price">
+            {product.currency}
+            {product.price}
+          </div>
+        </Col>
+      ));
+
+    // This will be more precise to show if it's still loading or failed as searching when we implement Fetching.
```
Comment (louisrli) [inline src/components/layout/nav/search/Search.jsx]: use TODO(github name) here too for any incomplete code, in general and in the future

File: src/components/layout/nav/search/Search.jsx [hunk 12]
```diff
@@ -1,7 +1,156 @@
...
+              value={searchQueryContainer}
+            />
+          </Col>
+          <Col xl={2} lg={2} md={2} className="col-3 searchButtonWrapper">
+            <Button onClick={sendSearchQuery}>Search</Button>
+          </Col>
+        </Col>
+        {searchQueryContainer ? (
+          <Col xl={6} lg={7} md={8} className="searchPreview">
+            {handleSearchPreview()}
```
Comment (louisrli) [inline src/components/layout/nav/search/Search.jsx]: see comments on omar's code about this one (the PR he submitted)

usually we use handle for something like an onclick event

this is simply a variable that you can assign to some JSX, you don't need to make it a function
`const searchPreview = (<div></div>)` mesela

File: src/components/layout/nav/search/Search.jsx [hunk 13]
```diff
@@ -1,7 +1,156 @@
...
+          </Col>
+        ) : (
+          showEmptySearchWarning && (
+            <span>You should type something to search.</span>
```
Comment (louisrli) [inline src/components/layout/nav/search/Search.jsx]: "Please type a search term."

Comment (louisrli) [reply src/components/layout/nav/search/Search.jsx]: you should is a bit forceful for the user :)

File: src/components/layout/nav/search/Search.scss [hunk 14]
```diff
@@ -0,0 +1,211 @@
...
+      -6px -6px 20px rgba(255, 255, 255, 0.25);
+    font-size: 16px;
+    margin-right: 20px;
+
+    &:hover {
+      cursor: pointer;
+    }
+
+    .title {
+      font-size: 1em;
```
Comment (louisrli) [inline src/components/layout/nav/search/Search.scss]: i'm pretty sure 1em is the default font-size?

File: src/components/layout/nav/search/Search.scss [hunk 15]
```diff
@@ -0,0 +1,211 @@
...
+    max-width: 70%;
+  }
+
+  .card {
+    padding-top: 10px;
+    padding-bottom: 10px;
+    background-color: #ffffff;
+    box-shadow: 1px 3px 20px rgba(0, 0, 0, 0.25),
+      -6px -6px 20px rgba(255, 255, 255, 0.25);
+    font-size: 16px;
```
Comment (louisrli) [inline src/components/layout/nav/search/Search.scss]: same with here, what's your global default font-size?

File: src/components/layout/nav/search/Search.scss [hunk 16]
```diff
@@ -0,0 +1,211 @@
...
+
+      // Do not show the third product at preview in this resolution.
+      .card:nth-last-of-type(1) {
+        display: none;
+      }
+    }
+  }
+}
+
+@media (max-width: $breakPointOfSmallDevices) {
```
Comment (louisrli) [inline src/components/layout/nav/search/Search.scss]: in general you should not have to handle every breakpoint manually like this, it makes it hard to maintain. this seems a bit wrong. there should only be two breakpoints that matter: hamburger menu or not hamburger menu

File: src/components/layout/nav/search/Search.scss [hunk 17]
```diff
@@ -0,0 +1,211 @@
...
+@media (max-width: breakPointPreviewBreaks) {
+  .searchPreview {
+    margin-top: 8px;
+  }
+}
+
+@media (max-width: $breakPointCardBreaks) {
+  .searchPreview {
+    .card {
+      font-size: 90%;
```
Comment (louisrli) [inline src/components/layout/nav/search/Search.scss]: for example: what's wrong here with simply using normal font-size for card breaks breakpoint? in general all these extra rules actually make your code more complicated and hard to maintain, i would recommend keeping it simpler

Comment (emrerdem1) [reply src/components/layout/nav/search/Search.scss]: I thought it is better to see where the `card` is located in but I can fix it of course.

File: src/components/layout/nav/search/Search.jsx (hunk 8 above)
Comment (emrerdem1) [reply src/components/layout/nav/search/Search.jsx]: It is an array of strings.

**Comment by emrerdem1** [comment]
PTAL

File: src/components/layout/nav/search/Search.jsx [hunk 18]
```diff
@@ -1,7 +1,151 @@
-import React from "react";
+import React, { useState } from "react";
+import Col from "react-bootstrap/Col";
+import Button from "react-bootstrap/Button";
+import Form from "react-bootstrap/Form";
+import Spinner from "react-bootstrap/Spinner";
+import "./Search.scss";
+
+// We agreed on switching to a 'common' mockDatabase after finishing merging HomePage succesfully.
```
Comment (louisrli) [inline src/components/layout/nav/search/Search.jsx]: nope, you have to do it first 🙂 generally we don't do temporary code like this because people can't be trusted to change it later, put it right from the beginning

File: src/components/layout/nav/search/Search.jsx [hunk 19]
```diff
@@ -1,7 +1,151 @@
...
+  },
+];
 
 const Search = () => {
-  return <div></div>;
+  // It is left empty because we want search to start as empty.
+  const [searchQueryContainer, setSearchQueryContainer] = useState("");
+  const [showEmptySearchWarning, setShowEmptySearchWarning] = useState(false);
+
+  const ONLY_FIRST_IMAGE_OF_PRODUCTS = 0;
```
Comment (louisrli) [inline src/components/layout/nav/search/Search.jsx]: outside of functions

Comment (louisrli) [reply src/components/layout/nav/search/Search.jsx]: like top of file after imports

File: src/components/layout/nav/search/Search.jsx [hunk 20]
```diff
@@ -1,7 +1,151 @@
...
+    isSearchInputEmpty();
+  };
+
+  // Set input values by onChange and show preview card only if there is something typed.
+  const handleSearchChanges = (e) => {
+    setSearchQueryContainer(e.target.value);
+    isSearchInputEmpty();
+  };
+
+  const isSearchInputEmpty = () =>
```
Comment (louisrli) [inline src/components/layout/nav/search/Search.jsx]: this variable name is wrong. isX suggests it's a boolean, but actually this function updates something

**Review by louisrli — CHANGES_REQUESTED** [review]

**Comment by emrerdem1** [comment]
PTAL

File: src/components/common/MockDatabase.js [hunk 21]
```diff
@@ -0,0 +1,68 @@
+export const MOCK_DATABASE = [
+  {
+    title: "Airmax",
+    images: [
```
Comment (louisrli) [inline src/components/common/MockDatabase.js]: do you need to pull. this shouldn't be a diff here

Comment (emrerdem1) [reply src/components/common/MockDatabase.js]: Sorry, I didn't understand. I already did pull and I couldn't see any `diff` in `Files changed` tab regarding this file neither. 
image

**Comment by emrerdem1** [comment]
PTAL

File: src/components/layout/nav/search/Search.scss [hunk 22]
```diff
@@ -0,0 +1,181 @@
+@import "../../../common/style.scss";
+
```
Comment (louisrli) [inline src/components/layout/nav/search/Search.scss]: i don't think this should be in search.scss, it affects all elements. it should be in app or index or something like that

File: src/components/layout/nav/search/Search.scss [hunk 23]
```diff
@@ -0,0 +1,181 @@
...
+        background-color: white;
+      }
+    }
+
+    button {
+      padding: 0;
+      height: 100%;
+      width: 100%;
+      background-size: 150%;
+      background-color: #8bc6ec;
```
Comment (louisrli) [inline src/components/layout/nav/search/Search.scss]: i believe this is the same color as burak. move it into common, use that variable here

**Review by louisrli — COMMENTED** [review]
genelde iyi gözüküyor ama css dosyaların fazla uzunmuş. galiba kuralların etkili değil, daha kısa olabilir. bundan sonra kısaltmaya çalış, mümkünse yani, ama en azından aklında tut

bu arada, yorum kullanımın çok gelişmiş :)

**Review by louisrli — APPROVED** [review]
i will approve for now, but please fix before submitting
